### PR TITLE
feat: Add "rest" function to Array

### PR DESCRIPTION
**Pull request by Efimero**
This is a very common idiom in Clojure that gets used a lot in reducers. It's nice to have.

### CODE REVIEW

File: core/Array.carp [hunk 1]
```diff
@@ -181,6 +181,10 @@ If the array is empty, returns `Nothing`")
   (defn suffix [xs start-index]
     (slice xs start-index (length xs)))
 
+  (doc rest "gets all but the first element from the array.")
+  (defn suffix [xs]
```
Comment (Efimero) [inline core/Array.carp:185]:
```suggestion
  (defn rest [xs]
```

Comment (Efimero) [inline core/Array.carp]:
```suggestion
  (defn rest [xs]
```

File: core/Array.carp [hunk 2]
```diff
@@ -181,6 +181,10 @@ If the array is empty, returns `Nothing`")
   (defn suffix [xs start-index]
     (slice xs start-index (length xs)))
 
+  (doc rest "gets all but the first element from the array.")
+  (defn rest [xs]
+    (slice xs 1 (length xs)))
```
Comment (hellerve) [inline core/Array.carp:186]: We could use `(suffix xs 1)` here to simplify it a little.

**Review by hellerve — APPROVED** [review]
Wow, I thought we had that function already! Good addition!

I left a little style nitpick, but I think we’re good to go.

File: core/Array.carp (hunk 2 above)
Comment (Efimero) [inline core/Array.carp:186]:
```suggestion
    (suffix xs 1))
```

**Pull request merged**